cdap-support-bundle/src/main/java/io/cdap/cdap/support/handlers/SupportBundleHttpHandler.java

```diff
@@ -111,4 +137,112 @@ public void createSupportBundle(HttpRequest request, HttpResponder responder,
       responder.sendString(HttpResponseStatus.OK, prevInProgressUUID);
     }
   }
+
+  /**
+   * Get the list of bundle status from status.json.
+   *
+   * @throws NotFoundException is thrown if status.json is not found
+   */
+  @GET
+  @Path("/support/bundles")
+  public void listSupportBundles(HttpRequest request, HttpResponder responder) throws Exception {
+    // ensure the user has authentication to create supportBundle
+    contextAccessEnforcer.enforceOnParent(EntityType.SUPPORT_BUNDLE, InstanceId.SELF, StandardPermission.LIST);
+    File baseDirectory = new File(cConf.get(Constants.SupportBundle.LOCAL_DATA_DIR));
+    if (!baseDirectory.exists()) {
+      LOG.debug("No content in Support Bundle.");
+      return;
+    }
+    List<SupportBundleOperationStatus> bundleOperationStatusList = new ArrayList<>();
+    DirUtils.listFiles(baseDirectory)
+      .stream()
+      .filter(file -> !file.isHidden() && file.isDirectory())
+      .forEach(uuidFile -> {
```

Comment:
Should we put UUID files in its own directory, e.g. `/<support-bundle-local-data-dir>/bundleUUIDs/`? Otherwise if we ever add a file to the base directory it'll get improperly listed as well.

Reply:
yes uuid has it's own directory take a look at supportbundlegenerator line: 123-126

Reply:
Oh I see, the UUID is a directory. If you don't forsee any other files being stored in this directory, this should be fine.